python/vespa/vespa/package.py

```diff
@@ -0,0 +1,412 @@
+import os
+from time import sleep
+from typing import List, Mapping, Optional
+from tempfile import TemporaryDirectory
+from pathlib import Path
+
+from jinja2 import Environment, PackageLoader, select_autoescape
+import docker
+
+from vespa.json_serialization import ToJson, FromJson
+
+
+class Field(ToJson, FromJson["Field"]):
+    def __init__(
+        self,
+        name: str,
+        type: str,
+        indexing: Optional[List[str]] = None,
+        index: Optional[str] = None,
+    ) -> None:
+        """
+        Object representing a Vespa document field.
+
+        :param name: Field name.
+        :param type: Field data type.
+        :param indexing: Configures how to process data of a field during indexing.
+        :param index: Sets index parameters. Content in fields with index are normalized and tokenized by default.
+        """
+        self.name = name
+        self.type = type
+        self.indexing = indexing
+        self.index = index
+
+    @property
+    def indexing_to_text(self) -> Optional[str]:
+        if self.indexing is not None:
+            return " | ".join(self.indexing)
+
+    @staticmethod
+    def from_dict(mapping: Mapping) -> "Field":
+        return Field(
+            name=mapping["name"],
+            type=mapping["type"],
+            indexing=mapping.get("indexing", None),
+            index=mapping.get("index", None),
+        )
+
+    @property
+    def to_dict(self) -> Mapping:
+        map = {"name": self.name, "type": self.type}
+        if self.indexing is not None:
+            map.update(indexing=self.indexing)
+        if self.index is not None:
+            map.update(index=self.index)
+        return map
+
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
+        return (
+            self.name == other.name
+            and self.type == other.type
+            and self.indexing == other.indexing
+            and self.index == other.index
+        )
+
+    def __repr__(self):
+        return "{0}\n{1}".format(self.__class__.__name__, str(self.to_dict))
+
+
+class Document(ToJson, FromJson["Document"]):
+    def __init__(self, fields: Optional[List[Field]] = None) -> None:
+        """
+        Object representing a Vespa document.
+
+        """
+        if not fields:
+            fields = []
+
+        self.fields = fields
```

Comment:
Just a different style, it might not be more clear: `self.fields = [] if not fields else fields`.